Default stylesheets revision + clean-up

MisterMakerNL opened the pull request:
This PR included 2 new stylesheets they are intended to be the only 2 included in main. It is a dark and a light version.
Any other stylesheets should be included as an addon.
![image](https://github.com/FreeCAD/FreeCAD/assets/29804962/9eeae39c-8bab-4632-83d5-705b467058c1)
![image](https://github.com/FreeCAD/FreeCAD/assets/29804962/8a4ba4a6-0dc9-4bab-858c-ff10e3ec019b)

![image](https://github.com/FreeCAD/FreeCAD/assets/29804962/6d97926f-64b0-4da8-87eb-b7335b61a1e0)
This also updated the look of Startpage 
![image](https://github.com/FreeCAD/FreeCAD/assets/29804962/bf5cc573-dfe5-4c05-a348-6866f9767b6d)

PaddleStroke commented:
Interesting. I seem to recall that Qt 6.5 was implementing a dark mode that would make things simpler. Is that taking advantage of that? Are we even using 6.5 btw?

MisterMakerNL commented:
@PaddleStroke this would indeed work very well with the 6.5 functionality as I understand Qt 6.5 can detect the dark style of windows and apply a dark theme. But it doesn't have a good dark theme out of the box. 
But would be more logical, if it did so maybe they have added one. QT docs aren't really complete most of the time.

MisterMakerNL commented:
Updated, already better than currently included stylesheets! So pretty much ready, just going to spend some time fixing up overlay.

![image](https://github.com/FreeCAD/FreeCAD/assets/29804962/099cd1c6-3b36-424a-8a65-78c645ee037f)

MisterMakerNL commented:
Btw if people think this is not modern or ugly because of the purple gradients, then we should fix that in no-stylesheet too and also here. This is a black version of no-stylesheet including the not so popular gradients. (although the menu gradients are adjusted to the white text.)

Might be a good idea to adjust the background color to the same gradient as in the menu like this:
![image](https://github.com/FreeCAD/FreeCAD/assets/29804962/7f2fd21e-3c70-4276-bdb6-51af4482041b)
I updated the Qsint group which now looks like original.
I also fixed the overlay a bit. worked around the properties panel click true issue by making the background 0.01% transparency.
![image](https://github.com/FreeCAD/FreeCAD/assets/29804962/4dd58262-0920-4190-b32a-aa86f298f94a)

https://github.com/FreeCAD/FreeCAD/issues/13351

MisterMakerNL commented:
Added light theme 
![image](https://github.com/FreeCAD/FreeCAD/assets/29804962/74a3a997-8254-44fe-a4c0-7f49ef48f2b3)

MisterMakerNL commented:
Added overlay fixes:
![image](https://github.com/FreeCAD/FreeCAD/assets/29804962/26793925-e08b-4079-a3d6-4d8be9bc70fe)

MisterMakerNL commented:
It probably has some tiny bugs, I see one with the treeview having black arrows.
But it's still a lot better than current situation, so this can be reviewed

chennes commented:
Are these meant to replace the stylesheets used in the "Dark" and "Light" preference packs?

MisterMakerNL commented:
> Are these meant to replace the stylesheets used in the "Dark" and "Light" preference packs?

Yes, I'll have a look at making the preference packs too.

chennes commented:
@FreeCAD/design-working-group can you please take a look?

pierreporte commented:
@chennes Looking at the screenshots, they look pretty decent. The only problem is the background but it has already been discussed: needs #13980 to be fixed before. The themes may not be perfect but they are good enough IMO and they can always be improved.

MisterMakerNL commented:
The idea here is also to work from this as a base.
Also move no-stylesheet to dev features.
These 2 styles and the OS-style. I see as the 3 options, in Windows you would have only 2 styles. 
If you pick OS-style it would just pick dark or light on the case how you have set your windows theme.
I think this how most programs work.

Then OpenThemes and Prodark and Behave-dark, pre-installed as an addon. (as long as they are actively maintained.)
I do not plan to maintain darker dark dork modern and such. It's to much work for what it's worth in my opinion. So I'd suggest to remove them unless someone wants to maintain them. 
 have a backup as an addon, which will have the last version as was included in Freecad.

FEA-eng commented:
> in Windows you would have only 2 styles

The old classic (current default) and the new dark classic? Only those would be shown in the first start widget? Makes sense but maybe the preinstalled OpenThemes (which are great and IMO should be included in the default package as you said) should also be shown there to make it clear for all users that they are already available. They are so good that one of them (likely Light) could even become the new default style but it's not a must for me.

MisterMakerNL commented:
I don't think Opentheme is ready, although the preference pack and the backgrounds are.
I have issues with the Stylesheets, as I also explained on Discord.
To sum it up:

1. The compile system that is used to compile the QSS files is in my view unnecessary complicated. And will make it harder for anyone. Especially non-linux users. Like you need git that is pretty obvious and easy for any Linux user, but anyone else would have no clue where to start.
2. The compile workflow can easily be broken by anyone who doesn't realizes it is compiles by editing the QSS files directly.
3. It still has a lot of tiny bugs that these stylesheets don't have. 

I am also aware that this compile system has benefits, but I really don't think we need those. It's not worth it. 

Also these Stylesheets are not that different from Opentheme.

Only a couple of things that really make these different from Opentheme are there too keep the Freecad look, but can easily be removed/changed.

1. Tabs
2. The use of the purplisch gradients.

obelisk79 commented:
Regarding the use of sass for the stylesheets. I don't think it is that confusing or complicated at all. This is something that is extremely common in the web development world where stylesheets are commonplace.  

I agree however regarding the little bugs, I've been chipping away at those. Some are rather problematic to isolate.

kadet1090 commented:
IMHO These styles should be merged only if the rest of preinstalled are removed. I however don't like that even though the stylesheets are aiming to be "simple", the source files are almost 3000 lines of code! It is a lot of code to maintain and to understand.

FEA-eng commented:
@kadet1090 So you are against merging this now? What about the rest of the DWG? @maxwxyz?

MisterMakerNL commented:
He doesn't say he is against merging it, he rather have all the others be removed. 
Which I agree on. But I already made a PR about that which wasn't successful 
https://github.com/FreeCAD/FreeCAD/pull/11982

kadet1090 commented:
> He doesn't say he is against merging it, he rather have all the others be removed.

Yes, this is closer to my opinion. I'm in favor of merging this as long as we will get rid of other pre-installed themes because if we don't do that we add next 6k lines of qss that must be maintained and virtually no-one does that resulting in overall poor quality of included themes. 

We should focus on having two well maintained ones in the core and not a dozen. As long as we don't get rid of other I'm against merging another ones as I fear that they will end up like the rest.

PaddleStroke commented:
+1, we need 2 basic themes well maintained. The rest should be addon.

FEA-eng commented:
> +1, we need 2 basic themes well maintained. The rest should be addon.

I would say 3 - classic (current default - for those who would heavily complained if it was removed) and 2 modern ones (light and dark).

kadet1090 commented:
Yes, I think of classic as "no theme"

MisterMakerNL commented:
I did a lot of work today, also compiled it tested it.
Had no issues.
There is only a tiny overlay text color issues in the light theme.
PR is a bit big, although not that complicated as I only change code and didn't really added any.

maxwxyz commented on src/Gui/Stylesheets/images_classic/Arrow-double-LeftRight-black.png:
I have no idea about styles but is it wanted that they are tilted? Can we replace them with .svgs? I can help.

MisterMakerNL replied:
No svg anymore those give scaling issues. PNG seem to render way nicer in QT especially on weird shaped icons.
These icons definitely shouldn't be slightly rotated. 
I'll fix that!
I do have a big svg file where these icons are in and can easily be exported. Not sure where I should upload this.

MisterMakerNL replied:
I fixed this, although double arrows aren't used atm.

maxwxyz commented on src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:
Can I provide new ones for these 3? This is all an example for a Don't use according to the stylesheet. Will classic still has the same BG color?

MisterMakerNL replied:
Off-course the idea of these they show the colors used in the theme, but in the Freecad logo.
I do think they are too big.
Not sure why @chennes added the dimensions in the filename but since he knows way more about this then me I did the same.

MisterMakerNL replied:
![Theme selection](https://github.com/FreeCAD/FreeCAD/assets/29804962/05978fa4-4a82-4cf6-9bcd-40639eab9fb1)

maxwxyz replied:
no :) either full color icon or the mono one in white, blue or red on dark backgrounds.

sytabaresa replied:
hey @MisterMakerNL great changes!
Only one recommendation, when you use mono version, make the F transparent, looks better:
![image](https://github.com/FreeCAD/FreeCAD/assets/2219914/a4a7b365-0301-451e-a0f3-073176d6f9e1)

MisterMakerNL replied:
The colors are based on the colors of the stylesheet :)
So changing the colors will break the functionality!

maxwxyz replied:
Maybe we should not display colors with the icon but with e.g. a toolbar / task panel with the color. I was in favor of the schematic images you did.

MisterMakerNL replied:
> Maybe we should not display colors with the icon but with e.g. a toolbar / task panel with the color. I was in favor of the schematic images you did.

I thought showing off the new logo was way more fun!

maxwxyz replied:
but it's now just 3 logos which have nothing to do with the theme unfortunately.
Maybe your proposal without the colorful icons, just the basic things as a demo of how it will look.
![grafik](https://github.com/FreeCAD/FreeCAD/assets/6246609/0e281968-7a90-49c3-a141-a91c121c3854)

MisterMakerNL replied:
Something like this?

![classic](https://github.com/FreeCAD/FreeCAD/assets/29804962/f8595e84-5e8e-482e-9d3b-666239dff1f3)
![dark](https://github.com/FreeCAD/FreeCAD/assets/29804962/67dffd2d-7ef2-439f-b285-110f297059f8)
![light](https://github.com/FreeCAD/FreeCAD/assets/29804962/5d399c62-db97-48bf-ae5a-4abbab9f48d3)

maxwxyz replied:
I would remove the colorful icons in the top left. not sure about preselect and select color at the same time but it's definitely better than the logo version. But will the light background be light or does it stay almost the same as dark?

MisterMakerNL replied:
Light background is gonna be Light as soon as we fix the sketch lines color issue.

MisterMakerNL replied:
![Theme selection](https://github.com/FreeCAD/FreeCAD/assets/29804962/9d704161-dacf-43e5-b534-814afb6ab634)
This is the svg file if you want to have a go at it.

MisterMakerNL commented:
> Do the default theme have control on the standard part color? The current one has too much contrast against the light background and not enough against the dark background.

I took all the preference-packs colors from OpenTheme, it seems that the light version isn't as far developed also the dark version. And needs some more work. 

I btw don't turn this pr into a draft because merging it would make Freecad still look better and will get me better/more bug reports.

pierreporte commented:
Do you prefer keeping the color as is and have a separate issue opened after merging? Hopefully it can be fixed in time.

MisterMakerNL commented:
> Do you prefer keeping the color as is and have a separate issue opened after merging? Hopefully it can be fixed in time.

I am trying to align the pr with the sketcher fixes from @kadet1090 . Hopefully then it is just tiny things.

MisterMakerNL commented:
> Do you prefer keeping the color as is and have a separate issue opened after merging? Hopefully it can be fixed in time.

I am now so far that it's only small style bugs, bugs that you really have to encounter while using the program. And so I think it is wise to push this PR so I get more bug reports. Especially the switch to white background can cause some issues in workbenches.
@chennes Do you got some time too, this also changes the startpage a tiny bit.

MisterMakerNL commented:
This also now changes the title bar in windows to black if you have dark mode on. Code thanks to @kadet1090 !

pierreporte commented:
> I am now so far that it's only small style bugs, bugs that you really have to encounter while using the program. And so I think it is wise to push this PR so I get more bug reports. Especially the switch to white background can cause some issues in workbenches.

Understood. I’m waiting for merge then.

chennes commented:
Could you please resolve the conflicts @MisterMakerNL?

MisterMakerNL commented:
> Could you please resolve the conflicts @MisterMakerNL?

I think I fixed the issue on maingui.cpp the other merge issues are about files I removed not sure why these are issues and how to fix them.

chennes commented:
I'm away from a computer for the weekend, but I can try to help get this cleaned up when I am back.

PaddleStroke commented:
Youtuber mentioning the Theme issue : 
https://youtu.be/p8Pk1ayx6LQ?si=-EsV5F3azbJRI7Ld&t=119

MisterMakerNL commented:
Dark looks like this 
![image](https://github.com/FreeCAD/FreeCAD/assets/29804962/c8d11815-829f-434a-8dfc-69738566bf58)

adrianinsaval commented:
any reason why "behave"  was kept around here? shouldn't we move it to an addon too?

MisterMakerNL commented:
> any reason why "behave" was kept around here? shouldn't we move it to an addon too?

Because I tried removing it before and it didn't got approved since it's being maintained by @Syres916 . I didn't wanted to stop this PR for being merged, because we got into a discussion about this. I do understand why we should remove it, because it's confusing and the style doesn't fit with the others. I hope @Syres916 also realizes this and is willingly to move his theme to an addon.
